libraries/botbuilder-stdlib/tests/types.test.js

```diff
@@ -0,0 +1,186 @@
+        { label: 'func', assert: typeAssert.func, input: new Date(), throws: true },
+
+        { label: 'nil', assert: typeAssert.nil, input: null, throws: false },
+        { label: 'nil', assert: typeAssert.nil, input: undefined, throws: false },
+        { label: 'nil', assert: typeAssert.nil, input: false, throws: true },
+
+        { label: 'number', assert: typeAssert.number, input: 10, throws: false },
+        { label: 'number', assert: typeAssert.number, input: false, throws: true },
+
+        { label: 'string', assert: typeAssert.string, input: 'hello', throws: false },
+        { label: 'string', assert: typeAssert.string, input: '', throws: false },
+        { label: 'string', assert: typeAssert.string, input: false, throws: true },
+    ];
+
+    testCases.forEach((testCase) => {
+        it(`typeAssert.${testCase.label} with ${JSON.stringify(testCase.input)} ${
+            testCase.throws ? 'throws' : 'does not throw'
+        }`, () => {
+            const test = () => testCase.assert(testCase.input, [testCase.label]);
+
+            if (testCase.throws) {
+                assert.throws(test);
+            } else {
+                assert.doesNotThrow(test);
+            }
+        });
+    });
+
+    describe('maybe', () => {
+        it('works', () => {
+            assert.doesNotThrow(() => typeAssert.maybeString('foo', []));
+            assert.doesNotThrow(() => typeAssert.maybeString(undefined, []));
+            assert.doesNotThrow(() => typeAssert.maybeString(null, []));
+            assert.throws(() => typeAssert.maybeString(true, []));
+        });
+
+        it('works for arrayOf', () => {
+            assert.doesNotThrow(() => typeAssert.maybeArrayOf(typeAssert.number)([1, 2, 3], []));
+            assert.doesNotThrow(() => typeAssert.maybeArrayOf(typeAssert.number)(undefined, []));
+            assertExt.throwsMessage(
+                () => typeAssert.maybeArrayOf(typeAssert.number)([1, 2, '3'], ['arr']),
+                '`arr.[2]` must be of type "number"'
+            );
+        });
+
+        it('works for instanceOf', () => {
+            assert.doesNotThrow(() => typeAssert.maybeInstanceOf('Error', Error)(new Error(), []));
+            assert.doesNotThrow(() => typeAssert.maybeInstanceOf('Error', Error)(undefined, []));
+            assertExt.throwsMessage(
+                () => typeAssert.maybeInstanceOf('Error', Error)('foo', ['err']),
+                '`err` must be an instance of "Error"'
+            );
+        });
+
+        it('works for oneOf', () => {
+            assert.doesNotThrow(() => typeAssert.maybeOneOf(tests.isNumber, tests.isString)(10, []));
+            assert.doesNotThrow(() => typeAssert.maybeOneOf(tests.isNumber, tests.isString)('11', []));
+            assertExt.throwsMessage(
+                () => typeAssert.maybeOneOf(tests.isNumber, tests.isString)(false, ['val']),
+                '`val` is of wrong type'
+            );
```

Comment:
I've noticed you're using this pattern a lot, which at a cursory glance could mean that the `maybeOneOf()` is state-ful.

For constant values provided from this library would it be acceptable to do `const maybeNumberOrString = typeAssert.maybeOneOf(tests.isNumber, tests.isString);`?

Reply:
We could (and should) provide some of these common `oneOf` combinations. I figured that would happen naturally as this style becomes more widely used.